Makefile

```diff
@@ -10,24 +10,27 @@ SDIR := src
 ODIR := obj
 BDIR := bin
 
+CC	     ?= gcc
+CXX      ?= g++
+CFLAGS   ?= -Wall -O3
+CXXFLAGS ?= -Wall -O3
+LDFLAGS  ?=
+
+CFLAGS	 += -MMD -MP -std=gnu99
+CXXFLAGS += -MMD -MP
+LDFLAGS  += -pthread
+
 # Set up flags depending on mode
 ifdef USERMODE
 PROG	 := yaha
-CCFLAGS  := -Wall -O3 -D COMPILE_USER_MODE -D BUILDNUM=$(BUILDNUM)
+CFLAGS   += -D COMPILE_USER_MODE -D BUILDNUM=$(BUILDNUM)
+CXXFLAGS += -D COMPILE_USER_MODE -D BUILDNUM=$(BUILDNUM)
```

Comment:
1. these should be added to a flag called `yaha_CPPFLAGS` (as they're preprocessor flags) and passed separately
2. preprocessor flags should never be split: use `-DCOMPILE_USER_MODE` over `-D COMPILE_USER_MODE`